src/config/enterprise.rs

```diff
@@ -601,6 +604,13 @@ pub(crate) fn report_configuration(
             opts,
         } = metadata;
 
+        // Get the machine hostname to avoid reporting the same configuration several time
+        // for a given instance of vector.
+        // If running on kubernetes, we take the pod name instead.
+        let hostname = env::var("VECTOR_SELF_POD_NAME").unwrap_or_else(|_| {
+            crate::get_hostname().expect("Couldn't read the current hostname.")
```

Comment:
No strong opinion on the fallback here as it seems like a product/business decision, which I'm fairly distant from these days.